wordpress-mobile/WordPress-Android
# Fix/9905 crash inflate exception binary xml file line

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/9905

**The main approach** for a potential fix is to increment `android material library` from `1.0.0` to `1.1.0-rc01`. 

Reason for that was because the codebase from the library was much improved. 

They changed a logic around crashing line from old release and general logic how drawables can be set. https://github.com/material-components/material-components-android/blob/cd59e98f7e2185ddb075ff0fc91f29765d562968/lib/java/com/google/android/material/textfield/TextInputLayout.java#L344 
(log -> https://sentry.io/share/issue/9eaadccac3be4c259943d72c46427792/). 

Thanks to @mzorz comment 👉 https://github.com/wordpress-mobile/WordPress-Android/issues/9905#issuecomment-514213827 I got inspired to make one more step in this fix.

**The second approach** is to use our own [xml](https://github.com/wordpress-mobile/WordPress-Android/commit/0f545724a56503906d3db9ab493a91ff4986bb21) which represents show/hide password icon in order to avoid a potential crash if icon can't be found in runtime.

**Note:** The new version of `android material library` forced some compile time issues : 

<img width="439" alt="Screen Shot 2019-07-23 at 3 43 16 PM" src="https://user-images.githubusercontent.com/28219092/61840486-57392600-ae5f-11e9-8359-87c4d8e9bfc8.png">

⬆️ this was resolved with adding `!!` and as I am not experienced in Kotlin, if someone has some better approach I am open for suggestions.

<img width="424" alt="Screen Shot 2019-07-23 at 3 44 25 PM" src="https://user-images.githubusercontent.com/28219092/61840487-57392600-ae5f-11e9-89c3-3b9a285586b3.png">


**To test:**

It's very hard to reproduce crash from the logs. However, one should check for potential regressions. 

In order to achieve that one should open and check every screen where `TextInputLayout` or any other component from `material` library is used. Especially`Enter your WordPress password` screen should be checked, where password visibility icon (show/hide password) shod work as before (expected).

Update release notes:
